lib/collection/src/collection_manager/optimizers/segment_optimizer.rs

```diff
@@ -358,27 +358,40 @@ pub trait SegmentOptimizer {
             PayloadFieldSchema,
         >::new()));
 
-        let proxies = optimizing_segments.iter().map(|sg| {
-            ProxySegment::new(
+        let mut proxies = Vec::new();
```

Comment:
Do we need vector of proxies? I expect that all field indexes should be the same in each `optimizing_segments`

Reply:
was wrong, it was a vector of proxies